### PR TITLE
Fix VER_ONLY hits an assertion

### PR DESCRIPTION
**Pull request by Xiaohanjlll**
Fix : #166 
Signed-off-by: Xiaohanjlll <hanx.xiao@intel.com>

### CODE REVIEW

File: spdm_emu/spdm_requester_emu/spdm_requester_spdm.c [hunk 1]
```diff
@@ -271,6 +271,10 @@ void *spdm_client_init(void)
         m_use_version = spdm_version >> SPDM_VERSION_NUMBER_SHIFT_BIT;
     }
 
+    if(m_exe_connection & EXE_CONNECTION_VERSION_ONLY){
+        goto done;
```
Comment (jyao1) [inline spdm_emu/spdm_requester_emu/spdm_requester_spdm.c:275]: why we have to skip?

Comment (Xiaohanjlll) [reply spdm_emu/spdm_requester_emu/spdm_requester_spdm.c:275]: `VER_ONLY `  Only the link version. I don't think the following code needs to be used

**Comment by Xiaohanjlll** [comment]
> Please describe what test you have run

I tested the other features of the `--exe_conn` `--exe_conn`  and default against each other.

File: spdm_emu/spdm_requester_emu/spdm_requester_spdm.c (hunk 1 above)
Comment (jyao1) [reply spdm_emu/spdm_requester_emu/spdm_requester_spdm.c:275]: I do not get it.
Why the following code needs not to be used?

**Comment by jyao1** [comment]
> > Please describe what test you have run
> 
> I tested the other features of the `--exe_conn` `--exe_conn` and default against each other.

Have you tested the command according to the bug description - `spdm_requester_emu --exe_conn "VER_ONLY" --meas_op "ALL" --pcap requester.pcap --exe_session "PSK,MEAS" --cap "ENCRYPT,MAC,PSK,HBEAT"` ?

**Comment by jyao1** [comment]
Please describe clearly what is the root cause of ASSERT .

**Comment by Xiaohanjlll** [comment]
> Please describe clearly what is the root cause of ASSERT .

The reason of the ASSERT is caused by   `--exe_conn "VER_ONLY"`

Only `libspdm_get_versionwill `be connected at this time

`spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_VERSION;`

data32  = spdm_context->connection_info.connection_state

ASSERT Judgment algorithm negotiation status update , `LIBSPDM_ASSERT(data32 == LIBSPDM_CONNECTION_STATE_NEGOTIATED);`

while ` -- exe_ Conn "VER_ONLY" ` Do not execute ` libspdm_ negotiate_ algorithms` 

The connection status is only `LIBSPDM_CONNECTION_STATE_AFTER_VERSION`

**Comment by jyao1** [comment]
Disagree the fix.

You can move spdm_load_negotiated_state() after libspdm_init_connection(), that will set state to LIBSPDM_CONNECTION_STATE_NEGOTIATED.

**Comment by jyao1** [comment]
please describe what test you have done.

**Comment by Xiaohanjlll** [comment]
> please describe what test you have done.

Test according to error description `- spdm_requester_emu --exe_conn "VER_ONLY" --meas_op "ALL" --pcap requester.pcap --exe_session "PSK,MEAS" --cap "ENCRYPT,MAC,PSK,HBEAT" `

And other characteristics of `--exe_conn VER_ONLY|DIGEST|CERT|CHAL|MEAS`

And whether the default status is successful

**Pull request merged**